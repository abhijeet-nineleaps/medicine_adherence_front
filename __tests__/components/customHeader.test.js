import React from "react";
import enableHooks from "jest-react-hooks-shallow";
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CustomHeader from "../../src/components/customHeader";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });
enableHooks(jest);
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
})); jest.mock("@react-native-google-signin/google-signin", () => ({
  default: jest.fn(),
  GoogleSignin: {
    Configure: () => ({}),
  }
}));
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useFocusEffect: jest.fn(),
  useEffect: jest.fn(),
}));
describe('customheader', () => {
  it("render component", () => {
    const wrapper = shallow(<CustomHeader />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it("test Touchable Opacity", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<CustomHeader touchFnc={mockFn} />);
    wrapper.find("#touch").props().onPress();
    expect(mockFn).toHaveBeenCalled;
  })
  it("test Button", () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<CustomHeader alertFnc={mockFn} />);
    wrapper.find("#alert").props().onPress();
    expect(mockFn).toHaveBeenCalled;
  })
});