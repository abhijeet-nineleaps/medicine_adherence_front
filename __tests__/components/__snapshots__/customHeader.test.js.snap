// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customheader renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
